src/linalg/basic/matrix.rs

```diff
@@ -49,27 +51,75 @@ pub struct DenseMatrixMutView<'a, T: Debug + Display + Copy + Sized> {
     column_major: bool,
 }
 
+// functional utility functions used across types
+fn is_valid_matrix_window(
+    mrows: usize,
+    mcols: usize,
+    vrows: &Range<usize>,
+    vcols: &Range<usize>,
+) -> bool {
+    debug_assert!(
+        vrows.end <= mrows && vcols.end <= mcols,
+        "The window end is outside of the matrix range"
+    );
+    debug_assert!(
+        vrows.start <= mrows && vcols.start <= mcols,
+        "The window start is outside of the matrix range"
+    );
+    debug_assert!(
+        // depends on a properly formed range
+        vrows.start <= vrows.end && vcols.start <= vcols.end,
+        "Invalid range: start <= end failed"
+    );
+
+    !(vrows.end <= mrows && vcols.end <= mcols && vrows.start <= mrows && vcols.start <= mcols)
+}
+fn start_end_stride(
```

Comment:
missing documentation

Reply:
I can describe each. I did not think it should be part of the public documentation as they are "internal/private".  So, `//` or `///`? What do you think?

Reply:
they should probably part of a trait, but I dont know which one, or a new.
Are they general or specific to a struct?

Reply:
@EdmundsEcho
IMO the new methods should be added in `impl DenseMatrix` after the last currently implemented method, for example:
```
    ///  <ADD PROPER DOC HERE> 
    fn is_valid_matrix_window(
        &self,
        mrows: usize,
        mcols: usize,
        vrows: &Range<usize>,
        vcols: &Range<usize>,
    ) -> bool
```
So that they can be called from the views constructors as:
```
if m.is_valid_matrix_window(mrows, mcols, &vrows, &vcols) { ... }
```

Reply:
I tried to add documentation using `///` but could not see in the final output of the docs.  So, I expanded on the description using `//`.  Secondly, I went through the rust-doc documentation.  It's not clear how to document the functions that are not part of a struct nor trait.  What I have seen in other libs is essentially what I have done.  Keep them local, comment for internal use only.